### PR TITLE
Fat 202 mod sender

### PR DESCRIPTION
**Pull request by nikolai991**

### CODE REVIEW

File: mod-sender/src/main/resources/domain/mod-sender/features/sender.feature [hunk 1]
```diff
@@ -9,4 +9,16 @@ Feature: Sender - message delivery
     Given path 'message-delivery'
     And request "{}"
     When method POST
-    Then status 422
\ No newline at end of file
+    Then status 422
+
```
Comment (OleksandrVidinieiev) [inline mod-sender/src/main/resources/domain/mod-sender/features/sender.feature:13]: Please add a test for case when requested user is not found (mod-users returns 404). See `SenderServiceImpl#lookupUser`.

File: mod-sender/src/main/resources/domain/mod-sender/features/sender.feature [hunk 2]
```diff
@@ -9,4 +9,32 @@ Feature: Sender - message delivery
...
+  Scenario: Should not found user
+    Given path 'message-delivery'
+    And request
+    """
+  { "notificationId":"db300321-d75a-48ce-87b0-7a387b3b21b2",
+    "recipientUserId":"undefined-db300321",
+    "messages":[
+     {"deliveryChannel":"email",
+      "from":"from","attachments":[]
```
Comment (OleksandrVidinieiev) [inline mod-sender/src/main/resources/domain/mod-sender/features/sender.feature:22]: add line feed

File: mod-sender/src/main/resources/domain/mod-sender/features/sender.feature [hunk 3]
```diff
@@ -9,4 +9,32 @@ Feature: Sender - message delivery
...
-    Then status 422
\ No newline at end of file
+    Then status 422
+
+  Scenario: Should not found user
+    Given path 'message-delivery'
+    And request
+    """
+  { "notificationId":"db300321-d75a-48ce-87b0-7a387b3b21b2",
+    "recipientUserId":"undefined-db300321",
```
Comment (OleksandrVidinieiev) [inline mod-sender/src/main/resources/domain/mod-sender/features/sender.feature:19]: please replace with proper UUID

File: mod-sender/src/main/resources/domain/mod-sender/features/sender.feature [hunk 4]
```diff
@@ -9,4 +9,32 @@ Feature: Sender - message delivery
     Given path 'message-delivery'
     And request "{}"
     When method POST
-    Then status 422
\ No newline at end of file
+    Then status 422
```
Comment (OleksandrVidinieiev) [inline mod-sender/src/main/resources/domain/mod-sender/features/sender.feature:12]: Please rename file to `messageDelivery.feature`

File: mod-sender/src/main/resources/domain/mod-sender/features/sender.feature (hunk 3 above)
Comment (nikolai991) [reply mod-sender/src/main/resources/domain/mod-sender/features/sender.feature]: Done

File: mod-sender/src/main/resources/domain/mod-sender/features/sender.feature (hunk 2 above)
Comment (nikolai991) [reply mod-sender/src/main/resources/domain/mod-sender/features/sender.feature]: Done

File: mod-sender/src/main/resources/domain/mod-sender/features/messageDelivery.feature [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    * callonce login testUser
+    * configure headers = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': 'application/json, text/plain' }
+
+  Scenario: Should return 422 when body is invalid
+    Given path 'message-delivery'
+    And request "{}"
+    When method POST
+    Then status 422
+
+  Scenario: Should not found user
```
Comment (OleksandrVidinieiev) [inline mod-sender/src/main/resources/domain/mod-sender/features/messageDelivery.feature:14]:
```suggestion
  Scenario: Should return 400 when user is not found
```

File: mod-sender/src/main/resources/domain/mod-sender/features/messageDelivery.feature [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    """
+    {
+      "notificationId": "db300321-d75a-48ce-87b0-7a387b3b21b2",
+      "recipientUserId": "1d413183-725d-45d5-a185-8c39916c5dd9",
+      "messages":
+          [
+            {   "deliveryChannel": "email",
+                "from":"from",
+                "attachments":[]
+            }
```
Comment (OleksandrVidinieiev) [inline mod-sender/src/main/resources/domain/mod-sender/features/messageDelivery.feature:26]:
```suggestion
            { 
                "deliveryChannel": "email",
                "from":"from",
                "attachments":[]
            }
```

File: mod-sender/src/main/resources/domain/mod-sender/features/messageDelivery.feature [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  @Undefined
+  Scenario: Should not fail when user contains additional properties
+    * print 'undefined'
+
+  @Undefined
+  Scenario: Should return no content and send email when request is valid
+    * print 'undefined'
+
+  @Undefined
+  Scenario: Should return bad request when delivery channel is not supported
```
Comment (OleksandrVidinieiev) [inline mod-sender/src/main/resources/domain/mod-sender/features/messageDelivery.feature:42]:
```suggestion
  Scenario: Should return 400 when delivery channel is not supported
```

File: mod-sender/src/main/resources/domain/mod-sender/features/messageDelivery.feature (hunk 6 above)
Comment (nikolai991) [reply mod-sender/src/main/resources/domain/mod-sender/features/messageDelivery.feature]: Done

File: mod-sender/src/main/resources/domain/mod-sender/features/messageDelivery.feature (hunk 5 above)
Comment (nikolai991) [reply mod-sender/src/main/resources/domain/mod-sender/features/messageDelivery.feature]: Done

File: mod-sender/src/main/resources/domain/mod-sender/features/messageDelivery.feature (hunk 7 above)
Comment (nikolai991) [reply mod-sender/src/main/resources/domain/mod-sender/features/messageDelivery.feature]: Done

**Review by OleksandrVidinieiev — APPROVED** [review]

**Pull request merged**